kernel/src/scan/log_replay.rs

```diff
@@ -81,27 +81,28 @@ impl DataVisitor for AddRemoveVisitor {
 // for `scan_row_schema` in scan/mod.rs! You'll also need to update ScanFileVisitor as the
 // indexes will be off
 pub(crate) static SCAN_ROW_SCHEMA: LazyLock<Arc<StructType>> = LazyLock::new(|| {
+    // Note that fields projected out of a nullable struct must be nullable
```

Comment:
hm, this feels weird (and i need to catch up on slack thread discussing this, sorry if I'm out of date) - we are losing information now on what is required in these schemas. do we have consumers of this schema that assume e.g. path is not null? how do we handle if path is missing now?

Reply:
this is an under specified comment, let me follow up more offline